Admin/DynamicListAdmin.php

```diff
@@ -45,20 +45,27 @@ public function __construct(ViewBuilderFactoryInterface $viewBuilderFactory, arr
 
     public function configureViews(ViewCollection $viewCollection): void
     {
-        $action = 'form-list';
-        // TODO handle multipage views
-
         foreach ($this->config as $parent => $sections) {
-            foreach ($sections as $key => $config) {
+            /** @var mixed[] $config */
+            foreach ($sections as $config) {
                 if (!$viewCollection->has($parent)) {
                     continue;
                 }
 
-                $name = $parent . '.' . $action . '-key';
+                $action = 'form-list';
+                if (isset($config['action'])) {
+                    $action = $config['action'];
+                }
+
+                $name = $parent . '.' . $action;
```

Comment:
Required to avoid a bc break of the view name:

```suggestion
                $name = $parent . '.' . $action . '-key';
```